test/common/json/json_sanitizer_test_util.cc

```diff
@@ -148,6 +155,25 @@ bool utf8Equivalent(absl::string_view a, absl::string_view b, std::string& diffs
   }
 }
 
+bool decodeEscapedJson(absl::string_view sanitized, std::string& decoded, std::string& errmsg) {
+  while (!sanitized.empty()) {
+    uint32_t hex;
+    if (sanitized.size() >= UnicodeEscapeLength &&
+        parseUnicode(sanitized.substr(0, UnicodeEscapeLength), hex)) {
+      if ((hex & 0xff) != hex) {
+        errmsg = absl::StrFormat("Unexpected encoding >= 256: %u", hex);
+        return false;
+      }
```

Comment:
The function is much easy to read. But I didn't get this. If the original input string contains 0xff, then hex here may be 0xff. Why we treat it an error?

Reply:
the predicate is identical to hex>=256. I don't know why I didn't just write it like that :)

changing!